worker/logforwarder/logforwarder.go

```diff
@@ -185,8 +186,13 @@ func (lf *LogForwarder) loop() error {
 	}
 
 	records := make(chan []logfwd.Record)
-	var stream LogStream
 	go func() {
+		var stream LogStream
+		defer func() {
+			if stream != nil {
+				_ = stream.Close()
```

Comment:
We need a test to confirm this.

Reply:
Nice spot, it was actually not always called for some reason (probably race condition between the go routine and the enclosing function.

I added some tests and put the logstream variable and the defer/close outside the goroutine, which fix this possible issue.

Reply:
(fixed)